types/jest/index.d.ts

```diff
@@ -667,25 +677,40 @@ declare namespace jest {
     interface Matchers<R, T> {
         /**
          * Ensures the last call to a mock function was provided specific args.
+         *
+         * Optionally, you can provide a type for the expected arguments via a generic.
+         * Note that the type must be either an array or a tuple.
          */
-        lastCalledWith(...args: any[]): R;
+        lastCalledWith<E extends any[]>(...args: E): R;
         /**
          * Ensure that the last call to a mock function has returned a specified value.
+         *
+         * Optionally, you can provide a type for the expected value via a generic.
+         * This is particuarly useful for ensuring expected objects have the right structure.
          */
-        lastReturnedWith(value: any): R;
+        lastReturnedWith<E>(value: E): R;
         /**
          * Ensure that a mock function is called with specific arguments on an Nth call.
+         *
+         * Optionally, you can provide a type for the expected arguments via a generic.
+         * Note that the type must be either an array or a tuple.
          */
-        nthCalledWith(nthCall: number, ...params: any[]): R;
+        nthCalledWith<E extends any[]>(nthCall: number, ...params: E): R;
         /**
          * Ensure that the nth call to a mock function has returned a specified value.
+         *
+         * Optionally, you can provide a type for the expected value via a generic.
+         * This is particuarly useful for ensuring expected objects have the right structure.
          */
-        nthReturnedWith(n: number, value: any): R;
+        nthReturnedWith<E>(n: number, value: E): R;
         /**
          * Checks that a value is what you expect. It uses `===` to check strict equality.
          * Don't use `toBe` with floating-point numbers.
+         *
+         * Optionally, you can provide a type for the expected value via a generic.
+         * This is particuarly useful for ensuring expected objects have the right structure.
          */
-        toBe(expected: any): R;
+        toBe<E>(expected: E): R;
```

Comment:
```suggestion
        toBe<E = any>(expected: E): R;
```

Explicit fallback rather than implicit